pkg/entrypoint/entrypointer.go

```diff
@@ -301,25 +325,7 @@ func (e Entrypointer) Go() error {
 	}
 
 	if e.ResultExtractionMethod == config.ResultExtractionMethodTerminationMessage {
-		// step artifacts
-		fp := filepath.Join(e.StepMetadataDir, "artifacts", "provenance.json")
-		artifacts, err := readArtifacts(fp, result.StepArtifactsResultType)
-		if err != nil {
-			logger.Fatalf("Error while handling step artifacts: %s", err)
-		}
-		output = append(output, artifacts...)
-
-		artifactsDir := pipeline.ArtifactsDir
-		// task artifacts
-		if e.ArtifactsDirectory != "" {
-			artifactsDir = e.ArtifactsDirectory
-		}
-		fp = filepath.Join(artifactsDir, "provenance.json")
-		artifacts, err = readArtifacts(fp, result.TaskRunArtifactsResultType)
-		if err != nil {
-			logger.Fatalf("Error while handling task artifacts: %s", err)
-		}
-		output = append(output, artifacts...)
+		e.appendArtifactOutputs(&output, logger)
```

Comment:
What's the reason for moving this code to a function? It does not help with testing and it is not re-used anywhere else.